src/services/LibraryService.ts

```diff
@@ -56,6 +57,34 @@ export class LibraryService implements Service<fhir4.Library> {
 
     const parsedParams = parseRequestSchema({ ...params, ...query }, PackageArgs);
 
-    return createLibraryPackageBundle(query, parsedParams);
+    const { libraryBundle } = await createLibraryPackageBundle(query, parsedParams);
+
+    return libraryBundle;
+  }
+
+  /**
+   * result of sending a POST or GET request to:
+   * {BASE_URL}/4_0_1/Library/$data-requirements or {BASE_URL}/4_0_1/Library/:id/$data-requirements
+   * creates a Library with all data requirements for the specified Library
+   * requires parameters id and/or url and/or identifier, but also supports version as supplemental (optional)
+   */
+  async dataRequirements(args: RequestArgs, { req }: RequestCtx) {
+    const params = gatherParams(req.query, args.resource);
+    validateParamIdSource(req.params.id, params.id);
+    const query = extractIdentificationForQuery(args, params);
+    const parsedParams = parseRequestSchema({ ...params, ...query }, LibraryDataRequirementsArgs);
+
+    logger.info(`${req.method} ${req.path}`);
+
+    const { libraryBundle, rootLibRef } = await createLibraryPackageBundle(query, parsedParams);
+
+    const { results } = await Calculator.calculateLibraryDataRequirements(libraryBundle, {
+      ...(params.periodStart && { measurementPeriodStart: params.periodStart }),
```

Comment:
Is there a reason why we use `params` for `calculateLibraryDataRequirements` but we use `parsedParams` when we calculate the data requirements in the `MeasureService`? Or should it also be `parsedParams` here?

Reply:
certainly `parsedParams` good catch!